### PR TITLE
Reimplement `mackerel_aws_integration` resource with terraform-plugin-framework

### PR DESCRIPTION
**Pull request by tosuke**
Part of #243 

Output from acceptance testing:

```
$ MACKEREL_EXPERIMENTAL_TFFRAMEWORK=1 make testacc TESTS='"AccMackerelAWSIntegration"'
TF_ACC=1 go test -v ./mackerel/... -run "AccMackerelAWSIntegration" -timeout 120m
2024/10/29 16:17:00 [INFO] mackerel: use terraform-plugin-framework based implementation
=== RUN   TestAccMackerelAWSIntegrationIAMRole
=== PAUSE TestAccMackerelAWSIntegrationIAMRole
=== RUN   TestAccMackerelAWSIntegrationCredentials
=== PAUSE TestAccMackerelAWSIntegrationCredentials
=== CONT  TestAccMackerelAWSIntegrationIAMRole
=== CONT  TestAccMackerelAWSIntegrationCredentials
--- PASS: TestAccMackerelAWSIntegrationCredentials (6.27s)
--- PASS: TestAccMackerelAWSIntegrationIAMRole (6.62s)
PASS
ok      github.com/mackerelio-labs/terraform-provider-mackerel/mackerel 7.743s
```

### CODE REVIEW

File: mackerel/resource_mackerel_aws_integration_test.go [hunk 1]
```diff
@@ -303,7 +303,6 @@ resource "mackerel_aws_integration" "foo" {
 
   ec2 {
     enable               = true
-    role                 = ""
```
Comment (tosuke) [inline mackerel/resource_mackerel_aws_integration_test.go:306]: The `role` field accepts a role ID(`$service_name: $role_name`) or null. Empty string is some kind of pathological case.

File: internal/provider/resource_mackerel_aws_integration.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+				},
+				"retire_automatically": schema.BoolAttribute{
+					Description: schemaAWSIntegrationServiceRetireAutomaticallyDesc,
+					Optional:    true,
+					Computed:    true,
+					Default:     booldefault.StaticBool(false),
+				},
+			},
+		},
+	}
```
Comment (tosuke) [inline internal/provider/resource_mackerel_aws_integration.go:232]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/5955898f77784ab62aa71fb604d48263492bdfea/mackerel/resource_mackerel_aws_integration.go?plain=1#L11-L69

File: internal/provider/resource_mackerel_aws_integration.go [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+				Computed:    true,
+				Default:     stringdefault.StaticString(""),
+			},
+			"excluded_tags": schema.StringAttribute{
+				Description: schemaAWSIntegrationExcludedTagsDesc,
+				Optional:    true,
+				Computed:    true,
+				Default:     stringdefault.StaticString(""),
+			},
+		},
```
Comment (tosuke) [inline internal/provider/resource_mackerel_aws_integration.go:321]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/5955898f77784ab62aa71fb604d48263492bdfea/mackerel/resource_mackerel_aws_integration.go?plain=1#L110-L150

**Review by ne-sachirou — APPROVED** [review]
🙆‍♀️

**Pull request merged**